visualization/app/codeCharta/util/algorithm/treeMapLayout/treeMapHelper.ts

```diff
@@ -203,13 +204,73 @@ export function getBuildingColor(node: CodeMapNode, { appSettings, dynamicSettin
 		return mapColors.flat
 	}
 
-	if (metricValue < dynamicSettings.colorRange.from) {
-		return mapColors.positive
-	}
-	if (metricValue > dynamicSettings.colorRange.to) {
-		return mapColors.negative
+	const positiveColorRGB = ColorConverter.convertHexToColorObject(mapColors.positive)
+	const neutralColorRGB = ColorConverter.convertHexToColorObject(mapColors.neutral)
+	const negativeColorRGB = ColorConverter.convertHexToColorObject(mapColors.negative)
+
+	switch (dynamicSettings.colorMode) {
+		case ColorMode.trueGradient: {
+			const middle = (dynamicSettings.colorRange.from + dynamicSettings.colorRange.to) / 2
+
+			if (metricValue <= middle) {
+				const factor = metricValue / middle
+				return ColorConverter.convertColorToHex(new Color().lerpColors(positiveColorRGB, neutralColorRGB, factor))
+			}
+
+			if (metricValue <= dynamicSettings.colorRange.max && metricValue > middle) {
+				const factor = (metricValue - middle) / (dynamicSettings.colorRange.max - middle)
+				return ColorConverter.convertColorToHex(new Color().lerpColors(neutralColorRGB, negativeColorRGB, factor))
+			}
+			break
+		}
+
+		case ColorMode.weightedGradient: {
+			const greenStart = Math.max(
+				dynamicSettings.colorRange.from - (dynamicSettings.colorRange.to - dynamicSettings.colorRange.from) / 2,
+				dynamicSettings.colorRange.from / 2
+			)
+
+			const yellowStart = 2 * dynamicSettings.colorRange.from - greenStart
+
+			const redStart = dynamicSettings.colorRange.to - (dynamicSettings.colorRange.to - dynamicSettings.colorRange.from) / 2
+
+			const redEnd = dynamicSettings.colorRange.to
+
+			if (metricValue <= greenStart) {
+				return mapColors.positive
+			}
+
+			if (metricValue >= greenStart && metricValue <= yellowStart) {
+				const factor = (metricValue - greenStart) / (yellowStart - greenStart)
+				return ColorConverter.convertColorToHex(new Color().lerpColors(positiveColorRGB, neutralColorRGB, factor))
+			}
+
+			if (metricValue >= yellowStart && metricValue <= redStart) {
+				return mapColors.neutral
+			}
+
+			if (metricValue >= redStart && metricValue <= redEnd) {
+				const factor = (metricValue - redStart) / (redEnd - redStart)
+				return ColorConverter.convertColorToHex(new Color().lerpColors(neutralColorRGB, negativeColorRGB, factor))
+			}
+
+			if (metricValue >= redEnd) {
+				return mapColors.negative
+			}
+			break
+		}
+
+		case ColorMode.absolute:
+			if (metricValue < dynamicSettings.colorRange.from) {
+				return mapColors.positive
+			}
+
+			if (metricValue > dynamicSettings.colorRange.to) {
+				return mapColors.negative
+			}
+			return mapColors.neutral
```

Comment:
This could be simplyfied
```suggestion
			if (metricValue < colorRange.from) {
				return mapColors.positive
			} else if (metricValue > colorRange.to) {
				return mapColors.negative
			}
			return mapColors.neutral
```

Reply:
EsLint dislikes that solution: "ESLint: Unnecessary 'else' after 'return'.(no-else-return)"

Reply:
Oh, you are right - my bad.